tests/product/yarn_slider/test_server_install.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+"""
+Tests for the server install sub command.
+"""
+
+from errno import ECONNREFUSED
+from socket import socket
+from StringIO import StringIO
+import time
+
+from prestoadmin.config import get_conf_from_json
+from prestoadmin.yarn_slider.config import SLIDER_CONFIG_PATH, PRESTO_PACKAGE
+
+from tests.product.base_product_case import BaseProductTestCase
+from tests.hdp_bare_image_provider import HdpBareImageProvider
+from tests.product.yarn_slider.slider_presto_installer import \
+    SliderPrestoInstaller
+
+HDFS_PORT = 8020
+
+
+class TestServerInstall(BaseProductTestCase):
+    def setUp(self):
+        super(TestServerInstall, self).setUp()
+        self.setup_cluster(HdpBareImageProvider(), self.PA_SLIDER_CLUSTER)
+        self.installer = SliderPrestoInstaller(self)
+        self.await_hdfs()
+
+    def await_hdfs(self):
+        start = time.clock()
+        ip_addr = self.cluster.get_ip_address_dict()[self.cluster.master]
+        while True:
+            try:
+                s = socket()
+                s.connect((ip_addr, HDFS_PORT))
+                break
+            except Exception as e:
+                s.close()
+                if e.errno == ECONNREFUSED:
+                    pass
+                else:
+                    raise
+            finally:
+                s.close()
+        end = time.clock()
+        duration = end - start
+        print 'Waited %.3f seconds for hdfs' % (duration)
+
+    def assert_uninstalled(self):
+        self.assertRaisesRegexp(OSError, 'No such file or directory',
+                                self.installer.assert_installed, self)
+
+    def get_config(self):
+        slider_master = self.installer._get_slider_master(self)
+        conf = self.get_file_content(slider_master, SLIDER_CONFIG_PATH)
+        conf_file = StringIO(conf)
+        return get_conf_from_json(conf_file)
```

Comment:
oh I see why.  But  once this is not actually a file, it's sort of forced to pretend that it is just so that you can use this method.  You could use json.loads() and operate directly on the string.